backend/googlephotos/googlephotos.go

```diff
@@ -654,24 +654,30 @@ func (f *Fs) listDir(ctx context.Context, prefix string, filter api.SearchFilter
 	if err != nil {
 		return nil, err
 	}
-	// Dedupe the file names
-	dupes := map[string]int{}
+
+	// Handle duplicate items
+	filenameCount := map[string]int{}
+	seenIDs := map[string]bool{}
+	newEntries := make([]fs.DirEntry, 0, len(entries))
 	for _, entry := range entries {
 		o, ok := entry.(*Object)
-		if ok {
-			dupes[o.remote]++
+		if !ok {
+			newEntries = append(newEntries, entry)
```

Comment:
Not sure if necessary to append in case ok is false or just continue is enough, but left it here to better match the original behaviour. Please let me know if this should be safe to remove.